2_munge/src/aggregate_inputs.R

```diff
@@ -116,6 +115,35 @@ aggregate_nwm <- function(ind_file, raw_ind_file, remake_file, sites_file, gd_co
         flow = if(n==24) mean(flow) else NA) %>%
       ungroup()
 
+  }else if(is_analysis){
+    valid_time_step = 1 # analysis goes back 3 hours?
+
+    agg_nwm <- streamflow %>%
+      as.data.frame() %>%
+      setNames(ref_time) %>% # columns are ref_time
+      dplyr::do(with(., {
+        ref_times = rep(as.POSIXct(colnames(.), tz='UTC'), each = nrow(.))
+        ref_dates = rep(as.Date(colnames(.), tz = 'UTC'), each = nrow(.))
+        n_ref_dates = ncol(.)
+        n_valid_times = input_raw$dim$time$len
+
+        out = data_frame(
+          ref_time = ref_times,
+          ref_date = ref_dates,
+          valid_time_hrs = rep(valid_time, n_ref_dates * n_sites),
+          flow = as.vector(as.matrix(.)),
+          site_no = rep(rep(site_nos, each = n_valid_times), times=n_ref_dates))
+      })) %>%
+      mutate(
+        valid_time = ref_time + as.difftime((valid_time_hrs - 3) * valid_time_step, units = 'hours'), # guessing on subtraction - Dave?
```

Comment:
@dblodgett-usgs here's the line I'd like your attention on - how do valid_time_hrs (values 1, 2, and 3) describe the relationship between reference time and valid time? the current code assumes that 1 means 2 hours before the reference time, 2 means 1 hour before the reference time, and 3 means the reference time. But I don't actually have any idea what the numbers really mean or where to look. Do you know?

Reply:
I'm having a hard time following what `valid_time_hrs` and the `n_ref_dates` are. But I think I see what's up. Up on line 75, you should be able to convert to posix the same as you do with reference time. That way your valid_time will just be a POSIXct the same as reference time?